docs/griptape-framework/drivers/prompt-drivers.md

```diff
@@ -347,87 +347,35 @@ agent = Agent(
-    `SageMakerLlamaPromptModelDriver` requires a tokenizer corresponding to a [Gated Model](https://huggingface.co/docs/hub/en/models-gated) on Hugging Face.
+Amazon Sagemaker Jumpstart provides a wide range of models with varying capabilities.
+This Driver has been primarily _chat-optimized_ models that have a [Huggingface Chat Template](https://huggingface.co/docs/transformers/en/chat_templating) available.
+If your model does not fit this use-case, we suggest sub-classing [AmazonSageMakerJumpstartPromptDriver](../../reference/griptape/drivers/prompt/amazon_sagemaker_jumpstart_prompt_driver.md) and overriding the [_to_model_input](../../reference/griptape/drivers/prompt/amazon_sagemaker_jumpstart_prompt_driver/#griptape.drivers.prompt.amazon_sagemaker_jumpstart_prompt_driver.AmazonSagemakerJumpstartPromptDriver._to_model_input) and [_to_model_params](../../reference/griptape/drivers/prompt/amazon_sagemaker_jumpstart_prompt_driver/#griptape.drivers.prompt.amazon_sagemaker_jumpstart_prompt_driver.AmazonSagemakerJumpstartPromptDriver._to_model_params) methods.
     
-    Make sure to request access to the [Meta-Llama-3-8B-Instruct](https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct) model on Hugging Face and configure your environment for hugging face use.
-
-```python title="PYTEST_IGNORE"
-import os
-from griptape.structures import Agent
-from griptape.drivers import (
-    AmazonSageMakerPromptDriver,
-    SageMakerLlamaPromptModelDriver,
-)
-from griptape.rules import Rule
-from griptape.config import StructureConfig
-
-agent = Agent(
-    config=StructureConfig(
-        prompt_driver=AmazonSageMakerPromptDriver(
-            endpoint=os.environ["SAGEMAKER_LLAMA_3_INSTRUCT_ENDPOINT_NAME"],
-            model=os.environ["SAGEMAKER_LLAMA_3_INSTRUCT_INFERENCE_COMPONENT_NAME"],
-            prompt_model_driver=SageMakerLlamaPromptModelDriver(),
-            temperature=0.75,
-        )
-    ),
-    rules=[
-        Rule(
-            value="You are a helpful, respectful and honest assistant who is also a swarthy pirate."
-            "You only speak like a pirate and you never break character."
-        )
-    ],
-)
-
-agent.run("Hello!")
-```
-
-##### Falcon
 
 ```python title="PYTEST_IGNORE"
 import os
 from griptape.structures import Agent
 from griptape.drivers import (
-    AmazonSageMakerPromptDriver,
+    AmazonSageMakerJumpstartPromptDriver,
     SageMakerFalconPromptModelDriver,
 )
 from griptape.config import StructureConfig
 
 agent = Agent(
     config=StructureConfig(
-        prompt_driver=AmazonSageMakerPromptDriver(
-            endpoint=os.environ["SAGEMAKER_FALCON_ENDPOINT_NAME"],
-            model=os.environ["SAGEMAKER_FALCON_INFERENCE_COMPONENT_NAME"],
-            prompt_model_driver=SageMakerFalconPromptModelDriver(),
+        prompt_driver=AmazonSageMakerJumpstartPromptDriver(
+            endpoint=os.environ["SAGEMAKER_LLAMA_3_INSTRUCT_ENDPOINT_NAME"],
+            model="meta-llama/Meta-Llama-3-8B-Instruct",
```

Comment:
We should add an `inference_component` parameter so that you can invoke endpoints with multiple models.

> Inference component name had been passed via the `model` parameter in `SageMakerLlamaPromptModelDriver`, but that was probably not the best name anyway since it isn't really related to the model id.

The "newer" SageMaker Studio JumpStart creates multiple model endpoints by default. Actually, I haven't even found a way to create single model endpoints in the newer studio.

Reply:
Updated